### PR TITLE
Made operation "bounds" take Predicate<Pixel>

### PR DESCRIPTION
**Pull request by u0ne**

### CODE REVIEW

File: jis-api/src/main/java/pl/edu/uj/JImageStream/model/Pixel.java [hunk 1]
```diff
@@ -1,21 +1,33 @@
 package pl.edu.uj.JImageStream.model;
 
-public class Pixel {
+import java.awt.Point;
 
-    //todo add x y, integrate into predicate
+public class Pixel extends Point{
 
     private int red;
```
Comment (jgprogramming) [inline jis-api/src/main/java/pl/edu/uj/JImageStream/model/Pixel.java]: could we hold colors already in int[] so getColors() method doesn't create new object every time it is called?

Comment (u0ne) [inline jis-api/src/main/java/pl/edu/uj/JImageStream/model/Pixel.java]: sure thing

File: jis-api/src/main/java/pl/edu/uj/JImageStream/api/transforms/AbstractImageTransform.java [hunk 2]
```diff
@@ -27,6 +28,11 @@ public AbstractImageTransform(BufferedImage bufferedImage, Filter filter, ColorC
 
     abstract protected void applyToPixels();
 
+    protected Pixel getPixel(int x, int y) {
```
Comment (jgprogramming) [inline jis-api/src/main/java/pl/edu/uj/JImageStream/api/transforms/AbstractImageTransform.java]: should be final

**Comment by u0ne** [comment]
done!

**Pull request merged**